gateway-service/src/main/java/org/zowe/apiml/gateway/security/login/saf/ZosAuthenticationProvider.java

```diff
@@ -0,0 +1,68 @@
+
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.security.login.saf;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.gateway.security.service.AuthenticationService;
+
+import java.util.Arrays;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class ZosAuthenticationProvider implements AuthenticationProvider, InitializingBean {
+    private PlatformUser platformUser = null;
+
+    private final AuthenticationService authenticationService;
+    private final String[] activeProfiles;
+
+    @Override
+    public Authentication authenticate(Authentication authentication) {
+        String userid = authentication.getName();
+        String password = authentication.getCredentials().toString();
+        PlatformReturned returned = getPlatformUser().authenticate(userid, password);
+
+        if ((returned == null) || (returned.isSuccess())) {
+            final String domain = "security-domain";
+            final String jwtToken = authenticationService.createJwtToken(userid, domain, "ltpa");
+            return authenticationService.createTokenAuthentication(userid, jwtToken);
+        } else {
+            throw new ZosAuthenticationException("Authentication failed: " + returned.toString(), returned);
+        }
+    }
+
+    private PlatformUser getPlatformUser() {
+        return platformUser;
+    }
+
+    @Override
+    public boolean supports(Class<?> authentication) {
+        return authentication.equals(UsernamePasswordAuthenticationToken.class);
+    }
+
+    @Override
+    public void afterPropertiesSet() {
+        if (platformUser == null) {
+            if (Arrays.asList(activeProfiles).contains("zos")) {
```

Comment:
Is this toggling appropriate? It will bloat our zos config.

Reply:
Our integration tests will always start with mock, never real impl.

Reply:
Agree. This is taken care of by CompoundAuthProvider.

Reply:
As we discussed, i would opt for dynamic recognition of SAF, and handle similar as passticket service. Also with a warning if SAF is not found, that SAF auth provides is being mocked and should not be used in production